app/src/main/java/ch/admin/bag/dp3t/MainActivity.java

```diff
@@ -125,9 +129,32 @@ private void checkIntentForActions() {
 			if (tracingViewModel.getTracingStatusInterface().wasContactReportedAsExposed()) {
 				gotoReportsFragment();
 			}
+		} else if (intent.getData() != null) {
+			checkValidCovidcodeIntent(intent.getData().toString());
 		}
 	}
 
+	private void checkValidCovidcodeIntent(String url) {
+		String[] fragmentSplit = url.split("#");
```

Comment:
We should make this change more robust. At the moment only https://swisscovid.ch/#123123123123 works, but not for example https://swisscovid.ch#123123123123 As we are not in control of the link generation, we should make this as robust as possible. I would suggest parsing it as an Uri and then looking at host, path and anchor individually.